contracts/smart-account/test/upgrades/BaseSmartAccountNew.sol

```diff
@@ -136,12 +117,33 @@ abstract contract BaseSmartAccountNew is IAccount, BaseSmartAccountErrors {
         }
     }
 
-    function init(address _owner, address _handler) external virtual;
+    /**
+     * ensure the request comes from the known entrypoint.
+     */
+    function _requireFromEntryPoint() internal view virtual {
+        require(
+            msg.sender == address(entryPoint()),
+            "account: not from EntryPoint"
+        );
+    }
 
-    function execTransaction(
-        Transaction memory _tx,
-        uint256 _batchId,
-        FeeRefund memory refundInfo,
-        bytes memory signatures
-    ) external payable virtual returns (bool success);
+    /**
+     * Validate the nonce of the UserOperation.
+     * This method may validate the nonce requirement of this account.
+     * e.g.
+     * To limit the nonce to use sequenced UserOps only (no "out of order" UserOps):
+     *      `require(nonce < type(uint64).max)`
+     * For a hypothetical account that *requires* the nonce to be out-of-order:
+     *      `require(nonce & type(uint64).max == 0)`
+     *
+     * The actual nonce uniqueness is managed by the EntryPoint, and thus no other
+     * action is needed by the account itself.
+     *
+     * @param nonce to validate
+     *
+     * solhint-disable-next-line no-empty-blocks
```

Comment:
can be removed as well

Reply:
yeah, if we choose to remove.

currently i have kept the description and also the method abstract - but commented like below until we make a decision to keep it.

`// function _validateNonce(uint256 nonce) internal view virtual;`

Reply:
I mean the solhint rule can be removed

Reply:
ok. can remove the rule.